### PR TITLE
drag n drop bounding characters (pgsql fix)

### LINKED ISSUE #14345: Drag and drop for Postgres should not include square brackets
- Azure Data Studio Version:
Version: 1.25.1 (system setup)
Commit: e7eacc32c0d20bd4da843320c3bfac231b308c98
Date: 2020-12-11T01:59:24.271Z
VS Code: 1.48.0
Electron: 9.3.0
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Windows_NT x64 10.0.19042

Steps to Reproduce:

1. Connect to a postgres server
2. use the object explorer drag and drop functionality to bring table and/or column names into the query editor
3. square brackets are placed around the object names, which is not valid postgres syntax

image

### PR DESCRIPTION
**Pull request by dzsquared**
This PR fixes #14345

Branches the object bounding characters by provider type - square bracket, quotation mark, and none (for safety on new providers)

escape-characters

### CODE REVIEW

**Comment by dzsquared** [comment]
let me know if you'd like changes to the tests in `src\sql\workbench\services\objectExplorer\test\browser\asyncServerTreeDragAndDrop.test.ts`

**Review by kburtram — APPROVED** [review]
We should be avoiding adding blocks of code that special-case providers.  These configurations should be proffered to the ADS shell through extensibility mechanisms.  Unfortunately, a lot of the recent customizations having been using these shortcuts, so it's fine for this bug fix, and we can clean up these code blocks later.

File: src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts [hunk 1]
```diff
@@ -102,18 +102,29 @@ export class ServerTreeDragAndDrop implements IDragAndDrop {
 			let providerName = this.getProviderNameFromElement(element);
 			if (providerName === 'KUSTO') {
 				finalString = element.nodeTypeId !== 'Function' && escapedName.indexOf(' ') > 0 ? `[@"${escapedName}"]` : escapedName;
-			} else {
+			} else if (providerName === 'MSSQL') {
 				finalString = escapedSchema ? `[${escapedSchema}].[${escapedName}]` : `[${escapedName}]`;
+			} else if (providerName === 'PGSQL') {
+				finalString = escapedSchema ? `"${escapedSchema}"."${escapedName}"` : `"${escapedName}"`;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts:108]: You shouldn't be using the escaped schema/name here - that's only for bracket-delimited names.

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]
I agree with what @kburtram said. I assume such a feature would be beyond the scope of what you're trying to do though so for the time being this is OK. But if you wanted to try and make this able to handle arbitrary providers I have some thoughts on that so chat with me offline.

**Comment by Charles-Gagnon** [comment]
And yes - it would be good to add tests for the different provider scenarios to validate that the escaping is doing what we expect.

**Comment by dzsquared** [comment]
I agree that it would be ideal to have an object to query text callback function as part of the data provider definition - the Kusto formatting for example is quite unique. I'll chat with @Charles-Gagnon but I suspect the implementation is beyond my skillset.

File: src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts [hunk 2]
```diff
@@ -102,18 +102,29 @@ export class ServerTreeDragAndDrop implements IDragAndDrop {
 			let providerName = this.getProviderNameFromElement(element);
 			if (providerName === 'KUSTO') {
 				finalString = element.nodeTypeId !== 'Function' && escapedName.indexOf(' ') > 0 ? `[@"${escapedName}"]` : escapedName;
-			} else {
+			} else if (providerName === mssqlProviderName) {
 				finalString = escapedSchema ? `[${escapedSchema}].[${escapedName}]` : `[${escapedName}]`;
+			} else if (providerName === 'PGSQL') {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts:107]: Let's make a constant for this as well and put it in sql/platform/connection/common/constants

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**